### ISSUE TITLE
ORA-01704 when generating large CLOB inline values in Oracle

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#6516]
The current inline serialisation of large `CLOB` and `BLOB` values in Oracle doesn't consider the `VARCHAR2(4000)` size limit (still the default in many Oracle 12c installations). We should generate a large `CLOB` expression, e.g. using:

```sql
SELECT to_clob('first-part ... after N characters, split') 
    || to_clob('... and concatenate the next part ...')
```

We can't use exactly `4000` characters, because that limit counts the number of bytes. So let's count those in `UTF-8`, and add a heuristic maximum of `2000` characters?

`BLOB` types suffer from a similar problem, which will be addressed separately: https://github.com/jOOQ/jOOQ/issues/14030

**Comment by lukaseder** [comment]
Moving the `BLOB` part to a separate issue, because it's harder to fix. We can already fix the `CLOB` part: https://github.com/jOOQ/jOOQ/issues/14030

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14031), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/14032)
